integrations/nginx/install.go

```diff
@@ -0,0 +1,349 @@
+package nginx
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/akitasoftware/akita-cli/printer"
+	"github.com/akitasoftware/akita-cli/telemetry"
+	"github.com/akitasoftware/go-utils/optionals"
+	"github.com/pkg/errors"
+)
+
+/* Top-level error to report to the user. */
+type InstallationError struct {
+	Wrapped error
+	Remedy  string
+}
+
+func (e *InstallationError) Unwrap() error {
+	return e.Wrapped
+}
+
+func (e *InstallationError) Error() string {
+	return fmt.Sprintf("Couldn't automatically install the Akita NGX module: %v", e.Wrapped)
+}
+
+var _ error = (*InstallationError)(nil)
+
+var (
+	unsupportedError = &InstallationError{
+		Wrapped: errors.New("The version of NGINX that is installed does not have a precompiled module."),
+		Remedy:  "Please try building the module from source, using the directions at https://github.com/akitasoftware/akita-nginx-module#building-the-module-from-source",
+	}
+
+	uninstalledError = &InstallationError{
+		Wrapped: errors.New("NGINX is not installed, or 'nginx' is not in the current path"),
+		Remedy:  "Please run this command on the machine where NGINX is installed.",
+	}
+
+	downloadError = &InstallationError{
+		Wrapped: errors.New("The precompiled module could not be downloaded from Akita."),
+		Remedy:  "Please download the module from https://github.com/akitasoftware/akita-nginx-module/releases/latest and complete the installation by copying it into the NGINX module directory.",
+	}
+
+	symlinkError = &InstallationError{
+		Wrapped: errors.New("The precompiled module was installed, but a symbolic link to it could not be created."),
+		Remedy:  "Please create a symbolic link in the module directory from ngx_http_akita_module.so to the installed module.",
```

Comment:
Would be nice if this provided a command to run, or if it said where the module directory is and where the module is installed.

Reply:
Fixed in https://github.com/akitasoftware/akita-cli/pull/198/commits/134d8a7f7cc5e16e380e5d0684feee305e86df2b

If the command was likely to work, probably the symlink would succeed anyway?  This really is a corner case where we're able to write to the directory, but not create a symlink within it. So I'm not sure what would cause that, maybe varying permissions on the directory and on the file itself (but I don't think that would do it.)